cloudbuild.yaml

```diff
@@ -291,27 +291,32 @@ steps:
       set -o pipefail
       pids=()
       cloudProducts=("generic" "gke-autopilot")
+      k8sVersions=("1.23" "1.24" "1.25")
+      clusterRegions=("us-east1" "us-west1" "us-central1")
+
       for cloudProduct in ${cloudProducts[@]}
       do
-        if [ $cloudProduct = generic ]
-        then 
-          featureWithGate="CustomFasSyncInterval=false&SafeToEvict=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
-          featureWithoutGate=""
-          testClusterLocation="us-west1-c"
-          testCluster="e2e-test-cluster"
-        else
-          featureWithGate="CustomFasSyncInterval=false&SafeToEvict=true&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
-          featureWithoutGate="SplitControllerAndExtensions=true"
-          testClusterLocation="us-west1"
-          testCluster="gke-autopilot-e2e-test-cluster-1-24"
-        fi
-        # TODO: Add more versions in the list, and generate the test cluster name in the loop rather than hardcode once the clusters are created
-        k8sVersions=("1.24")
-        for k8sVersion in ${k8sVersions[@]}
+        for i in ${!k8sVersions[@]}
         do
+          if [ $cloudProduct = gke-autopilot ] && [ ${k8sVersions[$i]} = 1.23 ]
+          then
+            continue
+          fi
+          if [ $cloudProduct = generic ]
+          then
+            featureWithGate="CustomFasSyncInterval=false&SafeToEvict=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
+            featureWithoutGate=""
+            testClusterLocation="${clusterRegions[$i]}-c"
```

Comment:
We should consider making the regular tests use regional clusters too. Maybe not as part of this PR, but something to consider since there is no cost difference between zonal & regional.

Reply:
Sounds good. I can make the change in the other PR where the test clusters are defined, and have a following up PR to change the CI steps once the clusters are created.